### ISSUE TITLE
Quickcheck warns about commented-out properties

### ISSUE THREAD

**Issue by andreasabel** [nick8325/quickcheck#52]
Applies to ver. 2.8.1
```haskell
import Test.QuickCheck
import Test.QuickCheck.All
{-
prop_old_freeVars_Term conf x = True
-}
-- Template Haskell hack to make the following $quickCheckAll work under ghc-7.8.
return [] 
-- | All tests as collected by 'quickCheckAll'.
tests :: IO Bool
tests = do
  putStrLn "Agda.TypeChecking.Free.Tests"
  $quickCheckAll
```
gives warning
```
/home/abel/agda/src/full/Agda/TypeChecking/Free/Tests.hs:129:3: Warning:
    Name prop_old_freeVars_Term found in source file but was not in scope
```
which turns into error with -Werror

**Comment by nick8325** [comment]
Hi,

This is definitely unfortunate. The problem is that Template Haskell gives (AFAIK) no way to find all the names in scope. So `quickCheckAll` works by reading the source file (!) and looking for lines that start with `prop_`.

The warning is mostly there so that if you forget the `return []` you get a warning instead of it just testing nothing. But in this case it gives a false positive. One workaround is to make the line not start with `prop_`, e.g. by using a single-line `--` comment instead of a multi-line one. I'll leave the bug open though until we have a fix.

**Comment by andreasabel** [comment]
A week ago or so I went looking for a tool that would strip comments from Haskell files.  Surprisingly, I did not find anything that is correct, fast, and works with unicode.  Should not be so hard to write that DFA.  Maybe Ana should give it as homework in her class.

**Comment by alanz** [comment]
If you use `getRichTokenStream`, filter out the comments, and then `showRichTokenStream` the GHC lexer will be able to help.

https://downloads.haskell.org/~ghc/latest/docs/html/libraries/ghc-7.10.1/GHC.html#g:33

**Comment by andreasabel** [comment]
Yes, but this is too heavy-weight (depending on GHC).

**Comment by asr** [comment]
The comments could be stripped using [haskell-src-exts](http://hackage.haskell.org/package/haskell-src-exts).

Example:

* [source code](https://github.com/haskell-suite/haskell-src-exts/blob/1a3a7b20e8a122537b075fe770fa0f57ac9134cf/tests/examples/HaddockComments.hs) with comments
* the [same source](https://github.com/haskell-suite/haskell-src-exts/blob/1a3a7b20e8a122537b075fe770fa0f57ac9134cf/tests/examples/HaddockComments.hs.prettyprinter.golden) code without comments

**Comment by nick8325** [comment]
That introduces rather a big dependency though. Maybe [haskell-lexer](https://hackage.haskell.org/package/haskell-lexer) could be an alternative.

**Comment by asr** [comment]
@andreasabel, please note that `{-# LANGUAGE TemplateHaskell #-}` is missing in the OP.
